src/backup_observer.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2024 Canonical Ltd.
```

Comment:
Minor: It's not a documented pattern but I think it would be nice to keep the Observer's file name/class the same as the integration in the metadata.yaml

Reply:
Renamed the integration from s3-backup to backup as suggested by @cbartz